src/main/java/net/mcreator/blockly/java/blocks/ProcedureCallBlock.java

```diff
@@ -116,30 +118,20 @@ public class ProcedureCallBlock implements IBlockGenerator {
 				Map<String, Object> dataModel = new HashMap<>();
 				dataModel.put("procedure", procedure.getName());
 				dataModel.put("dependencies", dependencies);
-
-				if (call_at) {
-					dataModel.put("x", xcode);
-					dataModel.put("y", ycode);
-					dataModel.put("z", zcode);
+				if (type.equals("call_procedure")) {
+					dataModel.put("depCount", depCount);
+					dataModel.put("names", names.keySet().stream().sorted().map(names::get).toArray(String[]::new));
+					dataModel.put("types", types.keySet().stream().sorted().map(types::get).toArray(String[]::new));
+					dataModel.put("args", args.keySet().stream().sorted().map(args::get).toArray(String[]::new));
 				}
 
-				if (master instanceof BlocklyToJava blocklyToJava
-						&& blocklyToJava.getEditorType() == BlocklyEditorType.COMMAND_ARG) {
-					if (type.equals("old_command")) {
-						master.append(
-								master.getTemplateGenerator().generateFromTemplate("_old_command.java.ftl", dataModel));
-					} else {
-						master.append(master.getTemplateGenerator()
-								.generateFromTemplate("_call_procedure.java.ftl", dataModel));
-					}
+				if (master instanceof BlocklyToJava btj && btj.getEditorType() == BlocklyEditorType.COMMAND_ARG

```

Comment:
Since the "old_command" block is only used in the command ME, couldn't you check only for the block type?

Reply:
I opened a PR on your branch, which solves this and should also improve generators code

Reply:
Is this PR more/less ready for merge after yours is merged here?

Reply:
Either way I would wait for other reviews before merging the PR

Reply:
✔️ Thanks a lot for the PR